### PR TITLE
Update 4ds template

### PR DESCRIPTION
**Pull request by pudingus**
Hi, I've updated the 4ds template based on my knowledge

Changed `byte` to `ubyte`
Correct handling of meshes with zero vertices
Renamed `unknown` to `weight` in HD2 skin
Handling of morph lods
Added variable names for morph unknowns
Added projectors, emitors, lights

### CODE REVIEW

File: 4ds.bt [hunk 1]
```diff
@@ -360,36 +368,52 @@ struct OCCLUDER
     FACE faces[numFaces];
 };
 
+struct PROJECTOR
```
Comment (RoadTrain) [inline 4ds.bt:371]: Out of curiosity, which models contain these frame types?

**Comment by RoadTrain** [comment]
Looks good, thanks for the contribution!

File: 4ds.bt (hunk 1 above)
Comment (pudingus) [reply 4ds.bt:371]: They are featured in Wings of War and Circus Empire, there are quite a lot of them
FRAME_PROJECTOR
\Circus Empire\Models\Interiors_efects\eff_model_svetla-vzor1.4ds

FRAME_EMITOR
\Wings Of War\models\Arras_prepady.4ds

FRAME_LIGHT and VISUAL_LITOBJECT
\Circus Empire\Models\Buildings\Exteriors\j_bigboard.4ds

These are just a few, I can compile a full list.

Comment (RoadTrain) [reply 4ds.bt:371]: Ah. I see! I thought there were some of them in Mafia :)

**Pull request merged**